### PR TITLE
Move Whisper types and protocol from nim-eth to this repo

### PR DESCRIPTION
**Pull request by oskarth**
Addresses https://github.com/status-im/nim-waku/issues/654 as part of refactor week

### CODE REVIEW

**Comment by oskarth** [comment]
@zah since you wrote the p2p macro IIRC, perhaps you have some ideas for how to do explicit error handling in above?

**Comment by oskarth** [comment]
After https://github.com/status-im/nim-eth/pull/374 has been merged the above should be easier, I'll look at it next week

**Review by jm-clius — APPROVED** [review]
LGTM

File: waku/v1/protocol/waku_bridge.nim [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
 import
   eth/p2p,
-  eth/p2p/rlpx_protocols/whisper_protocol,
+  #eth/p2p/rlpx_protocols/whisper_protocol,
```
Comment (kdeme) [inline waku/v1/protocol/waku_bridge.nim:15]: Left over comment

**Review by kdeme — COMMENTED** [review]
I'm rather indifferent with keeping whisper_protocol in nim-eth or moving it here. But if it gets moved, shouldn't the tests for whisper protocol also be moved (which are rather similar, I know)?

**Comment by oskarth** [comment]
> I'm rather indifferent with keeping whisper_protocol in nim-eth or moving it here. But if it gets moved, shouldn't the tests for whisper protocol also be moved (which are rather similar, I know)?

Right, fair point

**Pull request merged**